src/nspawn/nspawn.c

```diff
@@ -1051,6 +1058,54 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_SYSCALL_FILTER: {
+                        bool invert = false;
+
+                        if (isempty(optarg)) {
+                                arg_syscall_filter = set_free(arg_syscall_filter);
+                        }
+
+                        if (optarg[0] == '~') {
+                                invert = true;
+                                optarg++;
+                        }
+
+                        if (!arg_syscall_filter) {
+                                arg_syscall_filter = set_new(NULL);
+                                if (!arg_syscall_filter)
+                                        return log_oom();
+
+                                if (invert)
+                                        /* Apply default filter and the ones listed */
+                                        arg_syscall_filter_iswhitelist = true;
+                                else {
+                                        /* Apply default filter, exclude the ones listed */
+                                        arg_syscall_filter_iswhitelist = false;
+                                }
+                        }
+
+                        p = optarg;
+                        for (;;) {
+                                char *t = NULL;
+
+                                r = extract_first_word(&p, &t, ",", 0);
+                                if (r == 0)
+                                        break;
+                                if (r == -ENOMEM)
+                                        return log_oom();
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse syscall whitelist: %s", optarg);
+
+                                r = set_put(arg_syscall_filter, t);
+                                if (r == 0)
+                                        return 0;
```

Comment:
why are we returning here? IIUC, when `set_put()` returns 0 it means the key already exists so we can just ignore and continue the loop